tests/unit/Owncloud/OcisPhpSdk/GroupTest.php

```diff
@@ -0,0 +1,122 @@
+                "group_types" => ["aa"],
+                "members" => [],
+                // "membersodata_bind" => [],
+            ]],
+            [[
+                "id" => "id",
+                "description" => "desc",
+                "display_name" => "displayname",
+                "group_types" => [],
+                "members" => [],
+                // "membersodata_bind" => [],
+            ]],
+            [[
+                "id" => "id",
+                "description" => "desc",
+                "display_name" => "displayname",
+                "group_types" => [],
+                "members" => [],
+                // "membersodata_bind" => [],
+            ]],
+        ];}
+
+/**
+ * @dataProvider validGroupData
+ * @return void
+ */
+    public function testListGroup(array $data)
+    {
+        $libGroup = new Group($data);
+        $group = new SdkGroup($libGroup);
+        $this->assertInstanceOf(SdkGroup::class, $group);
+        $this->assertEquals($data["id"], $group->getId());
+        $this->assertEquals($data["description"], $group->getDescription());
+        $this->assertEquals($data["display_name"], $group->getDisplayName());
+        $this->assertSame($data["group_types"], $group->getGroupTypes());
+        $this->assertSame($data["members"], $group->getMembers());
+    }
+
+    private function InvalidGroupData(): array
+    {
+        return [
+            [[
+                "id" => null,
+                "description" => "",
+                "display_name" => "asd",
+                "group_types" => ["aa"],
+                "members" => [],
+            ]],
+            [[
+                "id" => "",
+                "description" => "a",
+                "display_name" => "as",
+                "group_types" => ["aa"],
+                "members" => [],
+            ]],
+        ];
+    }
+
+/**
+ * @dataProvider InvalidGroupData
+ * @return void
+ */
+    public function testInvalidIdInListGroup(array $data)
+    {
+        $this->expectException(InvalidResponseException::class);
+        $this->expectExceptionMessage("Invalid id returned for group '" . print_r($data["id"], true));
+        $libGroup = new Group($data);
+        $group = new SdkGroup($libGroup);
+    }
+
+    private function InvalidDisplayNameGroupData(): array
+    {
+        return [
+
+            [[
+                "id" => "as",
+                "description" => "a",
+                "display_name" => null,
+                "group_types" => ["aa"],
+                "members" => [],
+            ]],
+
+            [[
+                "id" => "sad",
+                "description" => "a",
+                "display_name" => "",
+                "group_types" => ["aa"],
+                "members" => [],
+            ]],
+        ];
+    }
+
+    /**
+     * @dataProvider InvalidDisplayNameGroupData
+     * @return void
+     */
+    public function testInvalidDidplayNameInListGroup(array $data)
```

Comment:
could be integrated into the test above, you just need to make the dataProvider return the expected message and then compare it to the real message